samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java

```diff
@@ -161,10 +200,31 @@ public void delete(K key) {
   public CompletableFuture<Void> deleteAsync(K key) {
     Preconditions.checkNotNull(key);
     writeMetrics.numDeletes.inc();
-    return execute(writeRateLimiter, key, writeFn::deleteAsync, writeMetrics.deleteNs)
-        .exceptionally(e -> {
-            throw new SamzaException(String.format("Failed to delete the record for " + key), (Throwable) e);
-          });
+
+    final long startNs = System.nanoTime();
+    CompletableFuture<Void> ioFuture;
```

Comment:
The same logic to compose the ioFuture w/ rate throttler and callbacks are repeated many times. Can we consolidate them?